x-pack/plugins/ml/public/application/datavisualizer/data_comparison/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,370 @@
+  );
+
+  const navigateToPath = useNavigateToPath();
+
+  useEffect(() => {
+    let unmounted = false;
+    const getMatchingDataView = async () => {
+      setDataViewMsg(undefined);
+      setFoundDataViewId(undefined);
+      if (!unmounted && referenceIndexPattern && productionIndexPattern) {
+        const indicesName = `${referenceIndexPattern},${productionIndexPattern}`;
+
+        const matchingDataViews = await dataViews.find(indicesName);
+
+        const timeFieldName =
+          Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+        if (Array.isArray(matchingDataViews) && matchingDataViews.length > 0) {
+          const foundDataView = matchingDataViews.find((d) => {
+            return d.timeFieldName === timeFieldName;
+          });
+
+          if (foundDataView) {
+            setFoundDataViewId(foundDataView.id);
+          } else {
+            setDataViewMsg(
+              i18n.translate(
+                'xpack.ml.dataDrift.indexPatternsEditor.hasDataViewWithDifferentTimeField',
+                {
+                  defaultMessage: `Found a data view matching pattern '{indexPattern}' but with a different time field. Creating a new data view to analyze data drift.`,
+                  values: { indexPattern: indicesName },
+                }
+              )
+            );
+          }
+        }
+      }
+    };
+
+    getMatchingDataView();
+
+    return () => {
+      unmounted = true;
+    };
+  }, [referenceIndexPattern, productionIndexPattern, timeField, dataViews]);
+  const createDataViewAndRedirectToDataComparisonPage = async (createAdHocDV = false) => {
+    // Create adhoc data view
+    const indicesName = `${referenceIndexPattern},${productionIndexPattern}`;
+
+    const timeFieldName =
+      Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+    let dataView;
+
+    if (!foundDataViewId) {
+      const modifiedDataViewName =
+        dataViewName === '' ? `${indicesName}-${timeFieldName}` : dataViewName;
+      if (canEditDataView && createAdHocDV === false) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = foundDataViewId ?? dataView?.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_COMPARISON,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  const firstSetOfSteps = [
+    {
+      title: 'Pick index pattern for reference data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
```

Comment:
Is there a reason why you don't get this error when the index pattern doesn't match any indices, like you see in the Data views page?

<img width="590" alt="image" src="https://github.com/elastic/kibana/assets/7405507/411fa628-c5b5-4732-898b-b65294740a24">

Can we disable the 'Analyze' buttons if the entered patterns don't match any indices?